cargo-screeps/src/copy.rs

```diff
@@ -0,0 +1,56 @@
+use std::path::{Path, PathBuf};
+
+use std::fs;
+
+use std::collections::HashSet;
+
+use failure;
+
+use config::Configuration;
+
+pub fn copy<P: AsRef<Path>>(root: P, config: &Configuration) -> Result<(), failure::Error> {
+    let root = root.as_ref();
+    let copy_config = config.copy.as_ref().ok_or_else(|| {
+        format_err!("must include [copy] section in configuration to deploy using copy")
+    })?;
+
+    let output_dir = copy_config.destination.join(&copy_config.branch);
+
+    fs::create_dir_all(&output_dir)?;
+
+    let target_dir = root.join("target");
+
+    let mut deployed: HashSet<PathBuf> = HashSet::new();
+
+    for entry in fs::read_dir(target_dir)? {
+        let entry = entry?;
+        let path = entry.path();
+
+        if let (Some(name), Some(extension)) = (path.file_name(), path.extension()) {
+            if extension == "js" {
```

Comment:
If at some point the user changes the configured file name, they'd end up with two "main" files and wasm files copied over.

Shouldn't that refer directly to the build config specific file names?

Reply:
If it should, it should change in both copy and upload. This preserves the current behavior of uploading files.

I.. think the purpose is to allow people to include extra javascript files if they want to?

Build will fail anyways if the project name is changed and `cargo clean` isn't run.